tests/search_memory_index.cpp

```diff
@@ -56,8 +56,11 @@ int search_memory_index(diskann::Metric& metric, const std::string& index_path,
   }
 
   bool filtered_search = false;
-  if (filter_label != "")
+  diskann::label filter_label_as_num = -1;
```

Comment:
same issue here: `diskann::label` is an unsigned: use `UINT_MAX` or some other appropriate value